lib/internal/process/promises.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+const promiseRejectEvent = process._promiseRejectEvent;
+const hasBeenNotifiedProperty = new WeakMap();
+const pendingUnhandledRejections = [];
+
+var scheduleMicrotasks;
+var NativeModule;
+module.exports = function initPromises(sMt, NM) {
+  scheduleMicrotasks = sMt;
+  NativeModule = NM;
+  return exports
+}
```

Comment:
I dunno how to get around some of this dependancy injection for now... Maybe add these to process and then remove them after boot?